File: LeetCodeSolutions/LeetCode_0035.py
class Solution:
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        for ans in range(len(nums)):
            if nums[ans] == target:
                return ans
            if nums[ans] > target:
                return ans
        return len(nums)

File: LeetCodeSolutions/LeetCode_0035.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l = 0, r = nums.size();
        if (r == 0) return 0;
        while (l < r){
            int mid = l + (r - l) / 2;
            if (nums[mid] >= target) r = mid;
            else  l = mid + 1;
        }
        return l;
    }
};